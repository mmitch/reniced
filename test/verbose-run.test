#!/bin/bash
set -e -o pipefail

# mock ps output
PATH=".:${PATH+:}${PATH}"
export PATH

# check if BSD::Resource is available
if perl -MBSD::Resource -e 1 2>/dev/null; then
    echo testing with BSD::Resource available >&2
    PRIO_PROCESS_SOURCE=BSD::Resource
else
    echo testing without BSD::Resource >&2
    PRIO_PROCESS_SOURCE=fallback
fi

# run test in no-op mode replacing placeholder with user id in output
../reniced -n -v test.conf 2>&1 | sed -e "s/${UID}/%%%USERID%%%/" -e "s/${PRIO_PROCESS_SOURCE}/%%%PRIO_PROCESS_SOURCE%%%/"
