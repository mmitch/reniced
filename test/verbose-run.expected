PRIO_PROCESS to 0 via fallback
rulefile: test.conf
3 rules read
ps commandline is: ps H -o lwp,comm --user 1000
5 processes read
would set priority of 201 to 5
nice set to 5: 201/foo --quick
would set priority of 2445 to 5
nice set to 5: 2445/foo --slow
would set IO priority for pid 19966 to class 3
ionice set to idle: 19966/bar
would set priority of 1234567 to 3
nice set to 3: 1234567/long-pid
would adjust OOM setting of pid 1234567 to -10
OOM adjust set to -10: 1234567/long-pid
